src/Nethermind/Nethermind.Facade/Filters/FilterStore.cs

```diff
@@ -172,64 +172,29 @@ private static TopicsFilter GetTopicsFilter(IEnumerable<Hash256[]?>? topics = nu
                 return SequenceTopicsFilter.AnyTopic;
             }
 
-            FilterTopic?[]? filterTopics = GetFilterTopics(topics);
             List<TopicExpression> expressions = new();
 
-            for (int i = 0; i < filterTopics?.Length; i++)
+            foreach (Hash256[]? group in topics)
             {
-                expressions.Add(GetTopicExpression(filterTopics[i]));
-            }
-
-            return new SequenceTopicsFilter(expressions.ToArray());
-        }
-
-        private static TopicExpression GetTopicExpression(FilterTopic? filterTopic)
-        {
-            if (filterTopic is not null)
-            {
-                if (filterTopic.Topic is not null)
+                if (group is null || group.Length == 0)
                 {
-                    return new SpecificTopic(filterTopic.Topic);
+                    expressions.Add(AnyTopic.Instance);
                 }
-
-                if (filterTopic.Topics?.Length > 0)
+                else if (group.Length == 1)
+                {
+                    expressions.Add(new SpecificTopic(group[0]));
+                }
+                else
                 {
-                    return new OrExpression(filterTopic.Topics.Select(
-                        static t => new SpecificTopic(t)).ToArray<TopicExpression>());
+                    expressions.Add(new OrExpression(group.Select(static t => new SpecificTopic(t)).ToArray<TopicExpression>()));
                 }
             }
 
-            return AnyTopic.Instance;
+            return new SequenceTopicsFilter(expressions.ToArray());
```

Comment:
We could avoid allocating `expressions` as a `List` and then doing `ToArray` as we can just allocate correct array size to begin with